src/EditorFeatures/Core.Wpf/ImageIdService.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Wpf;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Core.Imaging;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    [Shared]
+    [ExportWorkspaceService(typeof(IImageIdService), ServiceLayer.Host)]
+    internal class ImageIdService : IImageIdService
+    {
+        ImageId IImageIdService.GetImageId(Glyph glyph)
```

Comment:
Some context for @CyrusNajmabadi : Microsoft.VisualStudio.Core.Imaging.ImageId is a new struct mapping 1:1 to ImageMoniker, unfortunately we had to introduce it because ImageMoniker cannot be used in VS for Mac (as Mono doesn't support embedded interop types). It lives in MS.VS.CoreUtility dll as it's the lowest common layer between VS and VS for Mac.